src/modules/network_manager/network_manager_main.cpp

```diff
@@ -0,0 +1,241 @@
+/****************************************************************************
```

Comment:
Why is this structured this way?

This seems like it should a new file in the common stm32 board code and called from init function based on #define values for IP and MAC. A wrapper systemcmd could be added and a init script call could be done from rcS or read from SD card file to set mac, ip etc.

Reply:
@davids5 
Hi,

So some thoughts after reading your comment.

There is no need to add a file in order to use "#define" values , the define values are already used, the network manager only uses the parameters values, if someone wants to use the "#define" values then there's no need in network manager or similar.
The "#define" values are set in defconfig: 
-CONFIG_NSH_IPADDR 
-CONFIG_NSH_DRIPADDR
-CONFIG_NSH_NETMASK
-CONFIG_NSH_MACADDR

SD card can be changed or moved while parameters are per plane.
From my experience it is easy to use the Ethernet related values as parameters.
You can change it very easily, no need to take the SD off, change it and put it back.
If you're not sure what the IP is (or some other value) you can always look in a program like QGC, and again no need to read values from a file saved on the SD.
It's quick and easy to use (especially in case of more than one person using the same plane and more than one plane)  and I do think that the values should be changed through parameters.

Reply:
@garfieldG - Thank you for the perspective. I always am looking from a deeply embedded point of view at the cost of a feature set on resources. So it is good from me to have your perspective from a use-case perspective. Thank you for the changes for the defines.